backend/json2grpc/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+
+	atspb "json2grpc/spec"
+
+	"github.com/go-playground/validator/v10"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Send struct {
+	Sensor int32 `validate:"required,min=1,max=52"`
+}
+
+func main() {
+	var mutex = &sync.RWMutex{}
+	var sensor Send
+	var client atspb.AtsClient
+
+	//grpcサーバーとのコネクションの確立
+	address := "localhost:8888" //grpcサーバーのアドレス
+	conn, err := grpc.Dial(
+		address,
+
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Fatal("Connection failed")
+		return
+	}
+	defer conn.Close()
+	client = atspb.NewAtsClient(conn)
+
+	http.HandleFunc("/sensor", func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("Content-Type", "application/json")
+
+		switch r.Method {
+		case http.MethodGet:
+			mutex.RLock()
+
```

Comment:
[should]
このmutexは関数の中で呼ばれているので deferを用いて解放を書き直すことができます

```suggestion
defer mutex.RUnlock()
```

この場合だとEncoderがエラーを返したときにmutexが解放されなくてデッドロックに陥ります。